stl/inc/format

```diff
@@ -366,58 +366,347 @@ private:
+        return _Vis(_Arg._LLong_state);
+    case _Basic_format_arg_type::_ULLong_type:
+        return _Vis(_Arg._ULLong_state);
+    case _Basic_format_arg_type::_Bool_type:
+        return _Vis(_Arg._Bool_state);
+    case _Basic_format_arg_type::_Char_type:
+        return _Vis(_Arg._Char_state);
+    case _Basic_format_arg_type::_Float_type:
+        return _Vis(_Arg._Float_state);
+    case _Basic_format_arg_type::_Double_type:
+        return _Vis(_Arg._Double_state);
+    case _Basic_format_arg_type::_LDouble_type:
+        return _Vis(_Arg._LDouble_state);
+    case _Basic_format_arg_type::_Pointer_type:
+        return _Vis(_Arg._Pointer_state);
+    case _Basic_format_arg_type::_CString_type:
+        return _Vis(_Arg._CString_state);
+    case _Basic_format_arg_type::_String_type:
+        return _Vis(_Arg._String_state);
+    case _Basic_format_arg_type::_Custom_type:
+        return _Vis(_Arg._Custom_state);
+    default:
+        _STL_ASSERT(false, "Invalid basic_format_arg type");
+        return _Vis(0);
+    }
+}
+
+template <class _Context, class _Ty>
+/* consteval */ constexpr auto _Get_format_arg_storage_type() noexcept {
+    using _CharType = typename _Context::char_type;
+    if constexpr (is_same_v<_Ty, monostate>) {
+        return monostate{};
+    } else if constexpr (is_same_v<_Ty, _CharType>) {
+        return _CharType{};
+    } else if constexpr (is_same_v<_Ty, char> && is_same_v<_CharType, wchar_t>) {
+        return _CharType{};
+    } else if constexpr (signed_integral<_Ty> && sizeof(_Ty) <= sizeof(int)) {
+        return int{};
+    } else if constexpr (unsigned_integral<_Ty> && sizeof(_Ty) <= sizeof(unsigned int)) {
+        return static_cast<unsigned int>(42);
+    } else if constexpr (signed_integral<_Ty> && sizeof(_Ty) <= sizeof(long long)) {
+        return static_cast<long long>(42);
+    } else if constexpr (unsigned_integral<_Ty> && sizeof(_Ty) <= sizeof(unsigned int long long)) {
+        return static_cast<unsigned int long long>(42);
+    } else if constexpr (is_same_v<_Ty, bool>) {
+        return bool{};
+    } else if constexpr (is_same_v<_Ty, float>) {
+        return float{};
+    } else if constexpr (is_same_v<_Ty, double>) {
+        return double{};
+    } else if constexpr (is_same_v<_Ty, long double>) {
+        return static_cast<long double>(42);
+    } else if constexpr (is_same_v<_Ty, const void*>) {
+        return static_cast<const void*>(nullptr);
+    } else if constexpr (is_same_v<_Ty, const _CharType*>) {
+        return static_cast<const _CharType*>(nullptr);
+    } else if constexpr (is_same_v<_Ty, basic_string_view<_CharType>>) {
+        return basic_string_view<_CharType>{};
+    } else {
+        return basic_format_arg<_Context>::handle();
+    }
+}
+
+template <class _Context, class _Ty>
+/* consteval */ constexpr size_t _Get_format_arg_storage_size() noexcept {
+    return sizeof(_Get_format_arg_storage_type<_Context, _Ty>());
+}

```

Comment:
This could more cheaply be a variable template. (I'd be tempted to use an alias template for `decltype(_Get_format_arg_storage_type<_Meow, _Woof>())` also, but that's more a style thing.)